espnet/asr/pytorch_backend/asr.py

```diff
@@ -202,8 +287,10 @@ def __call__(self, batch, device):
             # because torch.nn.DataParellel can't handle it.
             xs_pad = {'real': xs_pad_real, 'imag': xs_pad_imag}
         else:
-            xs_pad = pad_list(
-                [torch.from_numpy(x).float() for x in xs], 0).to(device)
+            if evaluation:
+                xs_pad = pad_list([torch.from_numpy(x).float() for x in xs], 0).to(device)
+            else:
+                xs_pad = pad_list([specaug(torch.from_numpy(x).float()) for x in xs], 0).to(device)
```

Comment:
I think specaug(...) supports GPU tensor. How about this
```python
[specaug(torch.from_numpy(x).float().to(device)) for x in xs]
```
Or you can put `specaug` inside E2E (model) because E2E know it is training or evaluation and which device to use. It might be cleaner.

Reply:
To make it simple so far it didn't support masking. That is why I put it here so it can handle each sample individually.